dome9/resource_dome9_awp_aws_onboarding.go

```diff
@@ -0,0 +1,418 @@
+package dome9
+
+import (
+	"errors"
+	"fmt"
+	"github.com/dome9/dome9-sdk-go/dome9/client"
+	"github.com/dome9/dome9-sdk-go/services/awp_aws_onboarding"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-dome9/dome9/common/providerconst"
+	"log"
+	"strconv"
+	"strings"
+)
+
+func resourceAwpAwsOnboarding() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAWPAWSOnboardingCreate,
+		Read:   resourceAWPAWSOnboardingRead,
+		Update: resourceAWPAWSOnboardingUpdate,
```

Comment:
i am writing here (the correct location) : 
no currently only for few including agentless_account_settings (since it has an update api)

Reply:
I think in some cases an "update" triggers removal and re-creation of the resource. Do we want to make it behave this way here too? Would it work? Alternatively, can we fail the update if the user attempts to change an unsupported field? If so, how the user would cause re-creation? Is there some flag in terraform for such cases?

Reply:
Another option: reduce the parameters that the module will receive to only those that can be updated (not sure if it helps).

Reply:
we can't, since all of the needed parameters are required for the api call, without them we can't do proper api call

Reply:
I will attempt to implement when we try to update unsupported parameters, to fail the update flow
will check when the update triggers remocal and recreation and how if possible we can do recreation (to have the same behaviour)

Reply:
@chkp-alexgl Regarding your questions :

1. in Terraform, an update can indeed trigger a removal and re-creation of a resource. This is typically controlled by the **ForceNew: true** property in the resource schema. **If this property is set to true for a field, then changing that field will cause the resource to be destroyed and recreated.**
2. in our resource schema, the fields cloudguard_account_id and scan_mode have the **ForceNew: true property**. This **means if these fields are changed in the Terraform configuration, the resource will be destroyed and recreated.**
3. i can failed the update and preserve the behaviour for other fields by : 
 using CustomizeDiff function in the resource schema. This function allows us to add custom logic to check for changes and return an error if necessary.
4. i will implement this, **since removal and recreation behavior for the fields with ForceNew: true will still be preserved.** This is because the ForceNew property is handled by Terraform itself and is not affected by the CustomizeDiff function.

Reply:
Great! @bzemms please, decide whether we prefer to fail the update, or to mark those fields as ForceNew which would trigger re-creation. I don't fully understand all the cases when the user would change these things to have an opinion.